File: android-native/app/src/main/java/com/agentshell/feature/cron/CronJobEditorScreen.kt
package com.agentshell.feature.cron

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.unit.dp
import androidx.hilt.navigation.compose.hiltViewModel
import com.agentshell.data.model.CronJob
import java.util.UUID

private val schedulePresets = listOf(
    "Every hour" to "0 * * * *",
    "Every day" to "0 0 * * *",
    "Every week" to "0 0 * * 0",
    "Every month" to "0 0 1 * *",
    "Every 5 min" to "*/5 * * * *",
    "Every 30 min" to "*/30 * * * *",
    "Weekdays 9am" to "0 9 * * 1-5",
)

private val scheduleDescriptions = mapOf(
    "* * * * *" to "Every minute",
    "0 * * * *" to "Every hour at minute 0",
    "*/5 * * * *" to "Every 5 minutes",
    "*/10 * * * *" to "Every 10 minutes",
    "*/15 * * * *" to "Every 15 minutes",
    "*/30 * * * *" to "Every 30 minutes",
    "0 0 * * *" to "Every day at midnight",
    "0 9 * * *" to "Every day at 9:00 AM",
    "0 0 * * 0" to "Every Sunday at midnight",
    "0 0 1 * *" to "Monthly on the 1st at midnight",
    "0 0 * * 1-5" to "Weekdays at midnight",
    "0 9 * * 1-5" to "Weekdays at 9:00 AM",
)

private val llmProviders = listOf(
    "openai" to "OpenAI Compatible",
)

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun CronJobEditorScreen(
    existingJob: CronJob? = null,
    onNavigateUp: () -> Unit,
    viewModel: CronViewModel = hiltViewModel(),
) {
    val state by viewModel.state.collectAsState()
    val isEditing = existingJob != null

    var name by remember { mutableStateOf(existingJob?.name ?: "") }
    var schedule by remember { mutableStateOf(existingJob?.schedule ?: "0 * * * *") }
    var command by remember { mutableStateOf(existingJob?.command ?: "") }
    var emailTo by remember { mutableStateOf(existingJob?.emailTo ?: "") }
    var tmuxSession by remember { mutableStateOf(existingJob?.tmuxSession ?: "") }
    var workdir by remember { mutableStateOf(existingJob?.workdir ?: "") }
    var prompt by remember { mutableStateOf(existingJob?.prompt ?: "") }
    var llmModel by remember { mutableStateOf(existingJob?.llmModel ?: "gpt-5.4") }
    var llmProvider by remember { mutableStateOf(existingJob?.llmProvider ?: "openai") }
    var logOutput by remember { mutableStateOf(existingJob?.logOutput ?: false) }
    var showAdvanced by remember { mutableStateOf(false) }
    var showAiOptions by remember { mutableStateOf(existingJob?.prompt?.isNotEmpty() ?: false) }
    var envEntries by remember {
        mutableStateOf<List<Pair<String, String>>>(
            existingJob?.environment?.entries?.map { it.key to it.value } ?: emptyList()
        )
    }

    val hasAiJob = showAiOptions && workdir.isNotBlank() && prompt.isNotBlank()
    val hasCommand = command.isNotBlank()
    val isValid = name.isNotBlank() && schedule.isNotBlank() && (hasCommand || hasAiJob)
    val scheduleDesc = scheduleDescriptions[schedule] ?: "Custom schedule"

    fun save() {
        if (!isValid) return
        val envMap = envEntries.filter { it.first.isNotBlank() }.associate { it.first to it.second }
        val job = CronJob(
            id = existingJob?.id ?: UUID.randomUUID().toString(),
            name = name.trim(),
            schedule = schedule.trim(),
            command = command.trim(),
            enabled = existingJob?.enabled ?: true,
            lastRun = existingJob?.lastRun,
            nextRun = existingJob?.nextRun,
            createdAt = existingJob?.createdAt,
            emailTo = emailTo.trim().ifEmpty { null },
            logOutput = logOutput,
            tmuxSession = tmuxSession.trim().ifEmpty { null },
            environment = envMap.ifEmpty { null },
            workdir = if (showAiOptions) workdir.trim().ifEmpty { null } else null,
            prompt = if (showAiOptions) prompt.trim().ifEmpty { null } else null,
            llmProvider = if (showAiOptions) llmProvider else null,
            llmModel = if (showAiOptions) llmModel.trim().ifEmpty { null } else null,
        )
        if (isEditing) viewModel.updateJob(job) else viewModel.createJob(job)
        onNavigateUp()
    }

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text(if (isEditing) "Edit Cron Job" else "Create Cron Job") },
                navigationIcon = {
                    IconButton(onClick = onNavigateUp) {
                        Icon(Icons.Default.ArrowBack, contentDescription = "Back")
                    }
                },
                actions = {
                    TextButton(onClick = { save() }, enabled = isValid) {
                        Text(if (isEditing) "Update" else "Create")
                    }
                },
            )
        },
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues)
                .verticalScroll(rememberScrollState())
                .padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp),
        ) {
            OutlinedTextField(
                value = name,
                onValueChange = { name = it },
                label = { Text("Job Name") },
                placeholder = { Text("e.g., Backup Database") },
                modifier = Modifier.fillMaxWidth(),
            )

            // Schedule section
            Text("Schedule", style = MaterialTheme.typography.labelLarge)
            LazyRow(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                items(schedulePresets) { (label, value) ->
                    FilterChip(
                        selected = schedule == value,
                        onClick = { schedule = value },
                        label = { Text(label) },
                    )
                }
            }
            OutlinedTextField(
                value = schedule,
                onValueChange = { schedule = it },
                label = { Text("Cron Expression") },
                placeholder = { Text("* * * * * (minute hour day month weekday)") },
                textStyle = LocalTextStyle.current.copy(fontFamily = FontFamily.Monospace),
                modifier = Modifier.fillMaxWidth(),
                supportingText = { Text(scheduleDesc, color = MaterialTheme.colorScheme.outline) },
            )

            OutlinedTextField(
                value = command,
                onValueChange = { command = it },
                label = { Text("Command") },
                placeholder = { Text("/home/user/scripts/backup.sh") },
                textStyle = LocalTextStyle.current.copy(fontFamily = FontFamily.Monospace),
                modifier = Modifier.fillMaxWidth(),
                minLines = 2,
                maxLines = 4,
            )

            Row(verticalAlignment = Alignment.CenterVertically) {
                Button(
                    onClick = { viewModel.testCommand(command.trim()) },
                    enabled = command.isNotBlank() && !state.isTesting,
                ) {
                    if (state.isTesting) {
                        CircularProgressIndicator(modifier = Modifier.size(16.dp), strokeWidth = 2.dp)
                        Spacer(Modifier.width(8.dp))
                        Text("Testing...")
                    } else {
                        Icon(Icons.Default.PlayArrow, contentDescription = null, modifier = Modifier.size(18.dp))
                        Spacer(Modifier.width(8.dp))
                        Text("Test Command")
                    }
                }
            }

            state.testOutput?.let { output ->
                Surface(
                    color = MaterialTheme.colorScheme.surfaceVariant,
                    shape = MaterialTheme.shapes.small,
                ) {
                    Text(
                        text = output,
                        fontFamily = FontFamily.Monospace,
                        fontSize = MaterialTheme.typography.bodySmall.fontSize,
                        modifier = Modifier.padding(12.dp).fillMaxWidth(),
                    )
                }
            }

            // AI Options expandable section
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .clickable { showAiOptions = !showAiOptions }
                    .padding(vertical = 4.dp),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Icon(
                    if (showAiOptions) Icons.Default.KeyboardArrowDown else Icons.Default.KeyboardArrowRight,
                    contentDescription = null,
                    modifier = Modifier.size(20.dp),
                )
                Spacer(Modifier.width(4.dp))
                Text("AI Options", style = MaterialTheme.typography.bodyMedium)
            }

            if (showAiOptions) {
                OutlinedTextField(
                    value = workdir,
                    onValueChange = { workdir = it },
                    label = { Text("Working Directory") },
                    placeholder = { Text("/home/user/project") },
                    textStyle = LocalTextStyle.current.copy(fontFamily = FontFamily.Monospace),
                    modifier = Modifier.fillMaxWidth(),
                )
                OutlinedTextField(
                    value = prompt,
                    onValueChange = { prompt = it },
                    label = { Text("AI Prompt") },
                    placeholder = { Text("What should the AI agent do?") },
                    modifier = Modifier.fillMaxWidth(),
                    minLines = 3,
                    maxLines = 6,
                )

                var expanded by remember { mutableStateOf(false) }
                ExposedDropdownMenuBox(
                    expanded = expanded,
                    onExpandedChange = { expanded = it },
                    modifier = Modifier.fillMaxWidth(),
                ) {
                    OutlinedTextField(
                        value = llmProviders.firstOrNull { it.first == llmProvider }?.second ?: llmProvider,
                        onValueChange = {},
                        readOnly = true,
                        label = { Text("LLM Provider") },
                        trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(expanded) },
                        modifier = Modifier.menuAnchor().fillMaxWidth(),
                    )
                    ExposedDropdownMenu(expanded = expanded, onDismissRequest = { expanded = false }) {
                        llmProviders.forEach { (value, label) ->
                            DropdownMenuItem(
                                text = { Text(label) },
                                onClick = { llmProvider = value; expanded = false },
                            )
                        }
                    }
                }

                OutlinedTextField(
                    value = llmModel,
                    onValueChange = { llmModel = it },
                    label = { Text("LLM Model") },
                    placeholder = { Text("gpt-5.4") },
                    modifier = Modifier.fillMaxWidth(),
                )
            }

            // Advanced Options expandable section
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .clickable { showAdvanced = !showAdvanced }
                    .padding(vertical = 4.dp),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Icon(
                    if (showAdvanced) Icons.Default.KeyboardArrowDown else Icons.Default.KeyboardArrowRight,
                    contentDescription = null,
                    modifier = Modifier.size(20.dp),
                )
                Spacer(Modifier.width(4.dp))
                Text("Advanced Options", style = MaterialTheme.typography.bodyMedium)
            }

            if (showAdvanced) {
                OutlinedTextField(
                    value = emailTo,
                    onValueChange = { emailTo = it },
                    label = { Text("Email Output To") },
                    placeholder = { Text("user@example.com") },
                    modifier = Modifier.fillMaxWidth(),
                )

                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Text("Log command output to file", style = MaterialTheme.typography.bodyMedium)
                    Switch(checked = logOutput, onCheckedChange = { logOutput = it })
                }

                OutlinedTextField(
                    value = tmuxSession,
                    onValueChange = { tmuxSession = it },
                    label = { Text("Run in TMUX Session") },
                    placeholder = { Text("session-name") },
                    modifier = Modifier.fillMaxWidth(),
                )

                Text("Environment Variables", style = MaterialTheme.typography.labelLarge)
                Row(horizontalArrangement = Arrangement.spacedBy(6.dp)) {
                    listOf("PATH", "HOME", "TZ", "LANG").forEach { preset ->
                        AssistChip(
                            onClick = { envEntries = envEntries + (preset to "") },
                            label = { Text(preset) },
                        )
                    }
                }

                for (index in envEntries.indices) {
                    val (key, value) = envEntries[index]
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.spacedBy(8.dp),
                        verticalAlignment = Alignment.CenterVertically,
                    ) {
                        OutlinedTextField(
                            value = key,
                            onValueChange = { newKey ->
                                val updated = envEntries.toMutableList()
                                updated[index] = newKey to value
                                envEntries = updated
                            },
                            label = { Text("Key") },
                            textStyle = LocalTextStyle.current.copy(fontFamily = FontFamily.Monospace),
                            modifier = Modifier.weight(2f),
                            singleLine = true,
                        )
                        OutlinedTextField(
                            value = value,
                            onValueChange = { newValue ->
                                val updated = envEntries.toMutableList()
                                updated[index] = key to newValue
                                envEntries = updated
                            },
                            label = { Text("Value") },
                            textStyle = LocalTextStyle.current.copy(fontFamily = FontFamily.Monospace),
                            modifier = Modifier.weight(3f),
                            singleLine = true,
                        )
                        IconButton(onClick = {
                            val updated = envEntries.toMutableList()
                            updated.removeAt(index)
                            envEntries = updated
                        }) {
                            Icon(
                                Icons.Default.RemoveCircleOutline,
                                contentDescription = "Remove",
                                tint = MaterialTheme.colorScheme.error,
                            )
                        }
                    }
                }

                TextButton(
                    onClick = { envEntries = envEntries + ("" to "") },
                ) {
                    Icon(Icons.Default.Add, contentDescription = null)
                    Spacer(Modifier.width(4.dp))
                    Text("Add Variable")
                }
            }

            Spacer(Modifier.height(80.dp))
        }
    }
}

File: flutter/lib/features/cron/screens/cron_job_editor_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import '../../../data/models/cron_job.dart';
import '../providers/cron_provider.dart';

class CronJobEditorScreen extends ConsumerStatefulWidget {
  final CronJob? job;

  const CronJobEditorScreen({super.key, this.job});

  @override
  ConsumerState<CronJobEditorScreen> createState() =>
      _CronJobEditorScreenState();
}

class _CronJobEditorScreenState extends ConsumerState<CronJobEditorScreen> {
  final _formKey = GlobalKey<FormState>();
  late TextEditingController _nameController;
  late TextEditingController _scheduleController;
  late TextEditingController _commandController;
  late TextEditingController _emailController;
  late TextEditingController _tmuxController;
  late TextEditingController _workdirController;
  late TextEditingController _promptController;
  late TextEditingController _llmModelController;
  bool _logOutput = false;
  bool _showAdvanced = false;
  bool _showAiOptions = false;
  List<MapEntry<TextEditingController, TextEditingController>> _envEntries = [];
  String _llmProvider = 'openai';

  bool get isEditing => widget.job != null;

  static const schedulePresets = [
    {'label': 'Every hour', 'value': '0 * * * *'},
    {'label': 'Every day', 'value': '0 0 * * *'},
    {'label': 'Every week', 'value': '0 0 * * 0'},
    {'label': 'Every month', 'value': '0 0 1 * *'},
    {'label': 'Every 5 min', 'value': '*/5 * * * *'},
    {'label': 'Every 30 min', 'value': '*/30 * * * *'},
    {'label': 'Weekdays 9am', 'value': '0 9 * * 1-5'},
  ];

  static const llmProviders = [
    {'label': 'OpenAI Compatible', 'value': 'openai'},
  ];

  static final scheduleDescriptions = {
    '* * * * *': 'Every minute',
    '0 * * * *': 'Every hour at minute 0',
    '*/5 * * * *': 'Every 5 minutes',
    '*/10 * * * *': 'Every 10 minutes',
    '*/15 * * * *': 'Every 15 minutes',
    '*/30 * * * *': 'Every 30 minutes',
    '0 0 * * *': 'Every day at midnight',
    '0 9 * * *': 'Every day at 9:00 AM',
    '0 0 * * 0': 'Every Sunday at midnight',
    '0 0 1 * *': 'Monthly on the 1st at midnight',
    '0 0 * * 1-5': 'Weekdays at midnight',
    '0 9 * * 1-5': 'Weekdays at 9:00 AM',
  };

  @override
  void initState() {
    super.initState();
    _nameController = TextEditingController(text: widget.job?.name ?? '');
    _scheduleController = TextEditingController(
      text: widget.job?.schedule ?? '0 * * * *',
    );
    _commandController = TextEditingController(text: widget.job?.command ?? '');
    _emailController = TextEditingController(text: widget.job?.emailTo ?? '');
    _tmuxController = TextEditingController(
      text: widget.job?.tmuxSession ?? '',
    );
    _workdirController = TextEditingController(text: widget.job?.workdir ?? '');
    _promptController = TextEditingController(text: widget.job?.prompt ?? '');
    _llmModelController = TextEditingController(
      text: widget.job?.llmModel ?? 'gpt-5.4',
    );
    _llmProvider = widget.job?.llmProvider ?? 'openai';
    _logOutput = widget.job?.logOutput ?? false;
    _showAiOptions =
        widget.job?.prompt != null && widget.job!.prompt!.isNotEmpty;
    final env = widget.job?.environment ?? {};
    _envEntries = env.entries
        .map(
          (e) => MapEntry(
            TextEditingController(text: e.key),
            TextEditingController(text: e.value),
          ),
        )
        .toList();
  }

  @override
  void dispose() {
    _nameController.dispose();
    _scheduleController.dispose();
    _commandController.dispose();
    _emailController.dispose();
    _tmuxController.dispose();
    _workdirController.dispose();
    _promptController.dispose();
    _llmModelController.dispose();
    for (final entry in _envEntries) {
      entry.key.dispose();
      entry.value.dispose();
    }
    super.dispose();
  }

  String get scheduleDescription {
    return scheduleDescriptions[_scheduleController.text] ?? 'Custom schedule';
  }

  bool get isValid {
    final hasAiJob =
        _showAiOptions &&
        _workdirController.text.trim().isNotEmpty &&
        _promptController.text.trim().isNotEmpty;
    final hasCommand = _commandController.text.trim().isNotEmpty;
    return _nameController.text.trim().isNotEmpty &&
        _scheduleController.text.trim().isNotEmpty &&
        (hasCommand || hasAiJob);
  }

  void _save() {
    if (!isValid) return;

    final job = CronJob(
      id: widget.job?.id ?? DateTime.now().millisecondsSinceEpoch.toString(),
      name: _nameController.text.trim(),
      schedule: _scheduleController.text.trim(),
      command: _commandController.text.trim(),
      enabled: widget.job?.enabled ?? true,
      lastRun: widget.job?.lastRun,
      nextRun: widget.job?.nextRun,
      createdAt: widget.job?.createdAt ?? DateTime.now(),
      updatedAt: DateTime.now(),
      emailTo: _emailController.text.trim().isEmpty
          ? null
          : _emailController.text.trim(),
      logOutput: _logOutput,
      tmuxSession: _tmuxController.text.trim().isEmpty
          ? null
          : _tmuxController.text.trim(),
      environment: _buildEnvironmentMap(),
      workdir: _showAiOptions && _workdirController.text.trim().isNotEmpty
          ? _workdirController.text.trim()
          : null,
      prompt: _showAiOptions && _promptController.text.trim().isNotEmpty
          ? _promptController.text.trim()
          : null,
      llmProvider: _showAiOptions ? _llmProvider : null,
      llmModel: _showAiOptions && _llmModelController.text.trim().isNotEmpty
          ? _llmModelController.text.trim()
          : null,
    );

    if (isEditing) {
      ref.read(cronProvider.notifier).updateCronJob(job);
    } else {
      ref.read(cronProvider.notifier).createCronJob(job);
    }

    Navigator.pop(context);
  }

  void _testCommand() {
    if (_commandController.text.trim().isEmpty) return;
    ref.read(cronProvider.notifier).testCommand(_commandController.text.trim());
  }

  Map<String, String>? _buildEnvironmentMap() {
    final map = <String, String>{};
    for (final entry in _envEntries) {
      final key = entry.key.text.trim();
      final value = entry.value.text.trim();
      if (key.isNotEmpty) map[key] = value;
    }
    return map.isEmpty ? null : map;
  }

  void _addEnvEntry({String key = '', String value = ''}) {
    setState(() {
      _envEntries.add(
        MapEntry(
          TextEditingController(text: key),
          TextEditingController(text: value),
        ),
      );
    });
  }

  void _removeEnvEntry(int index) {
    setState(() {
      _envEntries[index].key.dispose();
      _envEntries[index].value.dispose();
      _envEntries.removeAt(index);
    });
  }

  @override
  Widget build(BuildContext context) {
    final cronState = ref.watch(cronProvider);

    return Scaffold(
      appBar: AppBar(
        title: Text(isEditing ? 'Edit Cron Job' : 'Create Cron Job'),
        actions: [
          TextButton(
            onPressed: isValid ? _save : null,
            child: Text(isEditing ? 'Update' : 'Create'),
          ),
        ],
      ),
      body: Form(
        key: _formKey,
        child: ListView(
          padding: const EdgeInsets.all(16),
          children: [
            TextField(
              controller: _nameController,
              decoration: const InputDecoration(
                labelText: 'Job Name',
                hintText: 'e.g., Backup Database',
                border: OutlineInputBorder(),
              ),
              onChanged: (_) => setState(() {}),
            ),
            const SizedBox(height: 16),
            Text('Schedule', style: Theme.of(context).textTheme.titleSmall),
            const SizedBox(height: 8),
            Wrap(
              spacing: 8,
              runSpacing: 8,
              children: schedulePresets.map((preset) {
                final isSelected = _scheduleController.text == preset['value'];
                return ChoiceChip(
                  label: Text(preset['label']!),
                  selected: isSelected,
                  onSelected: (_) {
                    setState(() {
                      _scheduleController.text = preset['value']!;
                    });
                  },
                );
              }).toList(),
            ),
            const SizedBox(height: 8),
            TextField(
              controller: _scheduleController,
              decoration: const InputDecoration(
                labelText: 'Cron Expression',
                hintText: '* * * * * (minute hour day month weekday)',
                border: OutlineInputBorder(),
              ),
              style: const TextStyle(fontFamily: 'monospace'),
              onChanged: (_) => setState(() {}),
            ),
            const SizedBox(height: 4),
            Text(
              scheduleDescription,
              style: Theme.of(
                context,
              ).textTheme.bodySmall?.copyWith(color: Colors.grey),
            ),
            const SizedBox(height: 16),
            TextField(
              controller: _commandController,
              decoration: const InputDecoration(
                labelText: 'Command',
                hintText: '/home/user/scripts/backup.sh',
                border: OutlineInputBorder(),
              ),
              style: const TextStyle(fontFamily: 'monospace'),
              maxLines: 3,
              onChanged: (_) => setState(() {}),
            ),
            const SizedBox(height: 8),
            Row(
              children: [
                ElevatedButton.icon(
                  onPressed: _commandController.text.trim().isEmpty
                      ? null
                      : _testCommand,
                  icon: cronState.isTesting
                      ? const SizedBox(
                          width: 16,
                          height: 16,
                          child: CircularProgressIndicator(strokeWidth: 2),
                        )
                      : const Icon(Icons.play_arrow, size: 18),
                  label: Text(
                    cronState.isTesting ? 'Testing...' : 'Test Command',
                  ),
                ),
              ],
            ),
            if (cronState.testOutput != null) ...[
              const SizedBox(height: 8),
              Container(
                padding: const EdgeInsets.all(12),
                decoration: BoxDecoration(
                  color: Colors.grey[900],
                  borderRadius: BorderRadius.circular(8),
                ),
                child: SelectableText(
                  cronState.testOutput!,
                  style: const TextStyle(
                    fontFamily: 'monospace',
                    fontSize: 12,
                    color: Colors.white,
                  ),
                ),
              ),
            ],
            const SizedBox(height: 16),
            InkWell(
              onTap: () => setState(() => _showAiOptions = !_showAiOptions),
              child: Row(
                children: [
                  Icon(
                    _showAiOptions
                        ? Icons.keyboard_arrow_down
                        : Icons.keyboard_arrow_right,
                    size: 20,
                  ),
                  const SizedBox(width: 4),
                  const Text('AI Options'),
                ],
              ),
            ),
            if (_showAiOptions) ...[
              const SizedBox(height: 16),
              TextField(
                controller: _workdirController,
                decoration: const InputDecoration(
                  labelText: 'Working Directory',
                  hintText: '/home/user/project',
                  border: OutlineInputBorder(),
                ),
                style: const TextStyle(fontFamily: 'monospace'),
              ),
              const SizedBox(height: 12),
              TextField(
                controller: _promptController,
                decoration: const InputDecoration(
                  labelText: 'AI Prompt',
                  hintText: 'What should the AI agent do?',
                  border: OutlineInputBorder(),
                ),
                maxLines: 4,
              ),
              const SizedBox(height: 12),
              DropdownButtonFormField<String>(
                value: _llmProvider,
                decoration: const InputDecoration(
                  labelText: 'LLM Provider',
                  border: OutlineInputBorder(),
                ),
                items: llmProviders.map((p) {
                  return DropdownMenuItem(
                    value: p['value'],
                    child: Text(p['label']!),
                  );
                }).toList(),
                onChanged: (value) {
                  if (value != null) {
                    setState(() => _llmProvider = value);
                  }
                },
              ),
              const SizedBox(height: 12),
              TextField(
                controller: _llmModelController,
                decoration: const InputDecoration(
                  labelText: 'LLM Model',
                  hintText: 'gpt-5.4',
                  border: OutlineInputBorder(),
                ),
              ),
            ],
            const SizedBox(height: 16),
            InkWell(
              onTap: () => setState(() => _showAdvanced = !_showAdvanced),
              child: Row(
                children: [
                  Icon(
                    _showAdvanced
                        ? Icons.keyboard_arrow_down
                        : Icons.keyboard_arrow_right,
                    size: 20,
                  ),
                  const SizedBox(width: 4),
                  const Text('Advanced Options'),
                ],
              ),
            ),
            if (_showAdvanced) ...[
              const SizedBox(height: 16),
              TextField(
                controller: _emailController,
                decoration: const InputDecoration(
                  labelText: 'Email Output To',
                  hintText: 'user@example.com',
                  border: OutlineInputBorder(),
                ),
              ),
              const SizedBox(height: 16),
              SwitchListTile(
                title: const Text('Log command output to file'),
                value: _logOutput,
                onChanged: (value) => setState(() => _logOutput = value),
                contentPadding: EdgeInsets.zero,
              ),
              const SizedBox(height: 8),
              TextField(
                controller: _tmuxController,
                decoration: const InputDecoration(
                  labelText: 'Run in TMUX Session',
                  hintText: 'session-name',
                  border: OutlineInputBorder(),
                ),
              ),
              const SizedBox(height: 16),
              const Text(
                'Environment Variables',
                style: TextStyle(fontWeight: FontWeight.w500),
              ),
              const SizedBox(height: 4),
              Wrap(
                spacing: 6,
                children: [
                  for (final preset in ['PATH', 'HOME', 'TZ', 'LANG'])
                    ActionChip(
                      label: Text(preset),
                      onPressed: () => _addEnvEntry(key: preset),
                    ),
                ],
              ),
              const SizedBox(height: 8),
              ..._envEntries.asMap().entries.map((entry) {
                final i = entry.key;
                final e = entry.value;
                return Padding(
                  padding: const EdgeInsets.only(bottom: 8),
                  child: Row(
                    children: [
                      Expanded(
                        flex: 2,
                        child: TextField(
                          controller: e.key,
                          decoration: const InputDecoration(
                            labelText: 'Key',
                            border: OutlineInputBorder(),
                            isDense: true,
                          ),
                          style: const TextStyle(fontFamily: 'monospace'),
                        ),
                      ),
                      const SizedBox(width: 8),
                      Expanded(
                        flex: 3,
                        child: TextField(
                          controller: e.value,
                          decoration: const InputDecoration(
                            labelText: 'Value',
                            border: OutlineInputBorder(),
                            isDense: true,
                          ),
                          style: const TextStyle(fontFamily: 'monospace'),
                        ),
                      ),
                      IconButton(
                        icon: const Icon(
                          Icons.remove_circle_outline,
                          color: Colors.red,
                        ),
                        onPressed: () => _removeEnvEntry(i),
                      ),
                    ],
                  ),
                );
              }),
              TextButton.icon(
                onPressed: _addEnvEntry,
                icon: const Icon(Icons.add),
                label: const Text('Add Variable'),
              ),
            ],
          ],
        ),
      ),
    );
  }
}
